### PR TITLE
Add client ErrExpiredToken error

### PR DESCRIPTION
**Pull request by karpovicham**
# Purpose
Add client errors to be able to recognize invalid token error response

# Implementation
- Create and expose `ErrExpiredToken`

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -14,8 +14,15 @@ import (
 const (
 	DevelopmentGateway = "https://api.development.push.apple.com"
 	ProductionGateway  = "https://api.push.apple.com"
+
+	// API errors
+	// https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns
+	errReasonExpiredProviderToken = "ExpiredProviderToken"
 )
 
+// Error when the token is stale and a new token should be generated.
```
Comment (mkorolyov) [inline client.go]:
```
type Error string

func (e Error) Error() string {
  return string(e)
}

const ErrExpiredToekn Error = "ExpiredProviderToken"
```

**Review by mkorolyov — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (karpovicham) [reply client.go]: Updated

File: client.go [hunk 2]
```diff
@@ -133,3 +129,15 @@ func setHeaders(r *http.Request, n *Notification) {
...
+
+// Map API error reason to client error if there is a reason.
+func apiErrorReasonToClientError(reason string) error {
+	switch reason {
+	case "":
+		return nil
+	case errReasonExpiredProviderToken:
+		return ErrExpiredToken
+	default:
+		return Error(reason)
```
Comment (karpovicham) [inline client.go]: @mkorolyov  Is it fine to return typed error here, or `fmt.Error` should be here?

File: Makefile [hunk 3]
```diff
@@ -1,3 +1,3 @@
 gen:
-	easyjson notification.go response.go
+	easyjson notification.go api.go
```
Comment (karpovicham) [inline Makefile:2]: just moved code from `response.go` to `api.go` (and Reason to be typed) to make `go-apns` and `go-fcm` structures look similar

File: client.go (hunk 2 above)
Comment (karpovicham) [reply client.go]: done with fmt.Error

**Review by mkorolyov — APPROVED** [review]

**Pull request merged**